### PR TITLE
Harden survey checking network access

### LINKED ISSUE #4463: android studio totally frozen with Flutter plugin v45
installing Flutter plugin v45 in android studio 192.7142 makes it totally frozen for about 20 seconds when switching editor dart tabs,
switching back Flutter plugin to v44 solved the problem.

### PR DESCRIPTION
**Pull request by pq**
Fixes #4463 

More eyes welcome.

/cc @stevemessick @devoncarew

### CODE REVIEW

File: src/io/flutter/survey/FlutterSurveyService.java [hunk 1]
```diff
@@ -9,68 +9,56 @@
...
   static FlutterSurvey getLatestSurveyContent() {
-    if (timeToUpdateCachedContent() || cachedSurvey == null) {
-      cachedSurvey = fetchSurveyContent();
+    // Update the cache daily.
+    if (timeToUpdateCachedContent()) {
+      ApplicationManager.getApplication().executeOnPooledThread(() -> {
+        cachedSurvey = fetchSurveyContent();
+      });
     }
     return cachedSurvey;
```
Comment (stevemessick) [inline src/io/flutter/survey/FlutterSurveyService.java:49]: Don't you need to wait for the Future on line 45 to complete?

Comment (pq) [reply src/io/flutter/survey/FlutterSurveyService.java:49]: No.  I've opted to make this async.  The assigned `cachedSurvey` will get picked up on a future editor tab switch or file open.  Precise timing isn't important.  (But not blocking **_is_**.)

Comment (stevemessick) [reply src/io/flutter/survey/FlutterSurveyService.java:49]: Got it. That might be worth explaining in the code.

Comment (pq) [reply src/io/flutter/survey/FlutterSurveyService.java:50]: Good idea.  Comment added.

**Pull request merged**